pkg/api/update_person_handler.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"encoding/json"
+
+	"github.com/darchlabs/api-example/pkg/person"
+	"github.com/go-playground/validator/v10"
+)
+
+type UpdatePersonHandlerInputs struct {
+	Name string `validate:"required"`
+	Age  int64  `validate:"required"`
+}
+
+type UpdatePersonHandler struct {
+	storage PersonStorage
+}
+
+func NewUpdatePersonHandler(ps PersonStorage) *UpdatePersonHandler {
+	return &UpdatePersonHandler{storage: ps}
+}
+
+func (*UpdatePersonHandler) Invoke(ctx *handlerCtx) *handlerRes {
+	//Get and parse id req param
+	id := ctx.r.FormValue("id")
+
+	// Validate body
+	validate := validator.New()
+	validate.Struct(ctx.ps)
+
+	// Define structure for parse body
+	body := &struct {
+		Person *person.Person `json:"person"`
+	}{} // TODO(ca): What does this second '{}' means?
```

Comment:
The `TODO` should have the initials of the writer. In this case `TODO(nb): ...`

Answering your question is because it's an inline definition, you are defining the struct and at the same time, you are initializing it. In short words, the second `{}` is an initialization with empty values for person=`nil`.

Reply:
Ok, good to know 👍